### DISCUSSION TITLE
How can I make a general middleware for authentication?

### DISCUSSION THREAD

**Discussion by vadim-su** [fastapi/fastapi#8164] [Questions] [answered]
How can I make a general middleware for authentication?
Is it possible to connect the middleware to a router or an endpoint?

**Comment by agundy** [comment, marked as answer]
One place you can do this is at the include_router level. See an example using dependencies for a router here: https://fastapi.tiangolo.com/tutorial/bigger-applications/#include-an-apirouter specifically the line `dependencies=[Depends(get_token_header)],`

**Comment by vadim-su** [comment]
Thanks for the idea!

**Comment by tiangolo** [comment]
Thanks for the help here @agundy ! :cake: :rocket: 

Thanks @deterok for reporting back and closing the issue. :tada:

**Comment by Evisolpxe** [comment]
Is it possible to remove endpoint dependencies singly in this router?

**Comment by Mause** [comment]
This conversation is over a year old, please open a new issue.
